py4DSTEM/process/diffraction/crystal_ACOM.py

```diff
@@ -1428,160 +1428,108 @@ def match_single_pattern(
+
+    # Generate an orix structure from the Crystal
+    atoms = [ Atom( element_symbols[Z-1], pos) for Z, pos in zip(self.numbers, self.positions)]
+
+    structure = Structure(
+        atoms=atoms,
+        lattice=Lattice(*self.cell),
+    )
+    
+    # Use pymatgen to get the symmetry
+    pg_structure = pgStructure(self.lat_real, self.numbers, self.positions, coords_are_cartesian=False)
+    pointgroup = SpacegroupAnalyzer(pg_structure).get_point_group_symbol()
+    
+    # If the structure has only one element, name the phase based on the element
+    if np.unique(self.numbers).size == 1:
+        name = element_symbols[self.numbers[0]-1]
+    else:
+        name = pg_structure.formula    
+    
+    # Generate an orix Phase to store symmetry
+    phase = Phase(
+        name=name,
+        point_group=pointgroup,
+        structure=structure,
+    )
+
+    xmap = CrystalMap(
+        rotations=rotations,
+        x=coords["x"],
+        y=coords["y"],
+        phase_list=PhaseList(phase),
+        prop={
+            "iq": orientation_map.corr[:, :, ind_orientation].ravel(),
+            "ci": orientation_map.corr[:, :, ind_orientation].ravel(),
+        },
+        scan_unit=pixel_units,
+    )
+
+    ckey = IPFColorKeyTSL(phase.point_group)
+
+    return (xmap, ckey) if return_color_key else xmap
 
 
 def save_ang_file(
     self,
     file_name,
     orientation_map,
-    ind_orientation = 0,
-    pixel_size: float = None,
-    ):
-    '''
-    This function outputs an ascii text file in the .ang format, containing 
-    the Euler angles of an orientation map. 
+    ind_orientation=0,
+    pixel_size=1.0,
+    pixel_units="px",
+):
+    """
+    This function outputs an ascii text file in the .ang format, containing
+    the Euler angles of an orientation map.
 
     Args:
+        file_name (str):                    Path to save .ang file.
         orientation_map (OrientationMap):   Class containing orientation matrices,
                                             correlation values, etc.
         ind_orientation (int):              Which orientation match to plot if num_matches > 1
         pixel_size (float):                 Pixel size, if known.
+        pixel_units (str):                  Units of the pixel size
 
     Returns:
         nothing
 
-    '''
+    """
 
-    num_dec = 4  # decimals of output numbers
+    from orix.io.plugins.ang import file_writer
 
-    if pixel_size is None:
-        pixel_size = 1.0
-    else:
-        pixel_size = np.array(pixel_size)
-
-    # Add orientation map index to file name
-    file_base = os.path.splitext(file_name)[0]
-    file_output = file_base + '_' + str(ind_orientation) + '.ang'
+    xmap = self.orientation_map_to_orix_CrystalMap(
+        orientation_map,
+        ind_orientation=ind_orientation,
+        pixel_size=pixel_size,
+        pixel_units=pixel_units,
+        return_color_key=False,
```

Comment:
This looks like it's hard coded to False, is there a workload where you'd want to save the `.ang` and return the `colour_key`?

Reply:
no